src/ensembl/src/shared/components/instant-download/instant-download-fetch/fetchForTranscript.ts

```diff
@@ -29,51 +32,88 @@ type Options = {
 };
 
 type FetchPayload = {
-  transcriptId: string;
+  genomeId: string;
   geneId: string;
+  transcriptId: string;
   options: Options;
 };
 
+type ProductGeneratingContextFragment = Pick<
+  ProductGeneratingContext,
+  'product' | 'cdna' | 'cds'
+>;
+
 export const fetchForTranscript = async (payload: FetchPayload) => {
   const {
+    genomeId,
     geneId,
     transcriptId,
     options: { transcript: transcriptOptions, gene: geneOptions }
   } = payload;
-  const urls = buildUrlsForTranscript(transcriptId, transcriptOptions);
+  const productGeneratingContext = await fetchSequenceChecksums({
+    genomeId,
+    transcriptId
+  });
+  const urls = buildUrlsForTranscript(
+    { geneId, productGeneratingContext },
+    transcriptOptions
+  );
+
   if (geneOptions.genomicSequence) {
-    urls.push(buildFetchUrl(geneId, 'genomicSequence'));
+    urls.push(buildFetchUrl({ geneId }, 'genomicSequence'));
   }
+
   const sequencePromises = urls.map((url) =>
-    fetch(url).then((response) => response.text())
+    fetch(url as string).then((response) => response.text())
```

Comment:
why the `as string`?

Reply:
This was because of the if statements in the `bulidFetchUrl` function having conditional returns (two `if` statements with no `else`) that resulted in a return type of `string | undefined`. I've made the changes in the function and also have removed the type assertion here.

Reply:
✅